runtime/parachains/src/system_token_manager.rs

```diff
@@ -269,33 +319,29 @@ pub mod pallet {
 				!SystemTokenOnParachain::<T>::contains_key(&wrapped_system_token),
 				Error::<T>::WrappedSystemTokenAlreadyRegistered
 			);
+			ensure!(
+				!ParaIdsBySystemToken::<T>::get(&system_token_id)
+					.unwrap()
+					.contains(&wrapped_system_token.clone().para_id),
+				Error::<T>::SameSystemTokenAlreadyRegistered
```

Comment:
I believe using match would be a better option than .unwrap to handle the result of ::get(), as the latter can cause a panic.

In the system_token_manager, I hope the rest of the unwrap calls can be replaced with match or other methods!

Reply:
Change to this
```
			let para_ids = {
				match ParaIdsBySystemToken::<T>::get(&system_token_id) {
					Some(para_ids) => para_ids,
					None => Default::default(),
				}
			};

			ensure!(
				!para_ids.contains(&wrapped_system_token.clone().para_id),
				Error::<T>::SameSystemTokenAlreadyRegistered
			);

```

Reply:
Or you can simply use 
```rust 
let para_ids = ParaIdsBySystemToken::<T>::get().map_or(Default::default(), |_| para_ids);
```

Reply:
Fix to
```
			let para_ids = ParaIdsBySystemToken::<T>::get(&system_token_id)
				.map_or(Default::default(), |para_id| para_id);

```